win32ss/gdi/ntgdi/freetype.c

```diff
@@ -2994,39 +2988,95 @@ static unsigned int get_bezier_glyph_outline(FT_Outline *outline, unsigned int b
     return needed;
 }
 
-static INT
-IntRequestFontSize(PDC dc, PFONTGDI FontGDI, LONG Width, LONG Height)
+static FT_Error
+IntRequestFontSize(PDC dc, PFONTGDI FontGDI, LONG lfWidth, LONG lfHeight)
 {
+    FT_Error error;
     FT_Size_RequestRec  req;
     FT_Face face = FontGDI->SharedFace->Face;
+    TT_OS2 *pOS2;
+    TT_HoriHeader *pHori;
+    FT_WinFNT_HeaderRec WinFNT;
+    LONG Ascent, Descent, Sum, EmHeight64;
 
-    if (Width < 0)
-        Width = -Width;
+    lfWidth = abs(lfWidth);
+    if (lfHeight == 0)
+    {
+        if (lfWidth != 0)
+            lfHeight = lfWidth;
+        else
+            lfHeight = -16;
```

Comment:
Is "lfHeight == 0 && lfWidth == 0" a case encountered in some real-life cases? As it seems to be an exception, maybe adding a DPRINT1 (or a WARN) for that case may help finding and understanding this pathological case.

Reply:
It's possible to encount lfHeight == 0 && lfWidth == 0. For example, a LOGFONT with zero-filled can be passed to CreateFontIndirect. It behaves like lfHeight == -16. DPRINT1 was added.